### ISSUE TITLE
map function not honouring Option.SUPPRESS_EXCEPTIONS

### ISSUE THREAD

**Issue by ghost** [json-path/JsonPath#612]
Hi, 
DocumentContext.map() is not honouring Option.SUPPRESS_EXCEPTIONS, while DocumentContext.read() is working fine

sample test - 

```
    @Test
    public void test() {
        Configuration config = Configuration.builder()
                .options(Option.SUPPRESS_EXCEPTIONS)
                .build();
        String json = "{\"1\":{\"2\":null}}";
        DocumentContext documentContext = JsonPath.using(config).parse(json);
        JsonPath query = JsonPath.compile("$.1.2.a.b.c");
        Assert.assertNull(documentContext.read(query));
        try {
            documentContext.map(query, (object, configuration) -> object);
        } catch (Exception e) {
            fail("got exception: " + e);
        }
    }
```
which fails with message - 
```
got exception: com.jayway.jsonpath.PathNotFoundException: Expected to find an object with property ['a'] in path $['1']['2'] but found 'null'. This is not a json object according to the JsonProvider: 'com.jayway.jsonpath.spi.json.JsonSmartJsonProvider'.
```

**Comment by CindyChow123** [comment]
Hi @Errorblade ,I would like to fix this bug.

**Comment by kallestenflo** [comment]
Fixed by #711
